### PR TITLE
Pull down environment files from GitHub if not present locally

### PR DESCRIPTION
**Pull request by bnemanich**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [x] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [x] Did you write any [tests](https://github.com/open-ce/open-ce/blob/main/tests/) to validate this change?  

## Description

Pulls down environment files from GitHub if not found locally. This allows the user to use commands such as:

`open-ce build env pytorch-env` and `open-ce build env --git_tag_for_env 1.1.3 opence-env`

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

**Comment by bethune-bryant** [comment]
What kind of error message gets shown if they make a typo in the env file name, is it useful enough to figure out they had a typo?

**Comment by bnemanich** [comment]
It is will always print out an info statement saying what file it is trying to download. If that fails, it will get the same error message that other failed fetches of URLs get:

```
open-ce build env --skip_build_packages xgbost-env
INFO: Unable to find xgbost-env locally. Attempting to use https://raw.githubusercontent.com/open-ce/open-ce-environments/main/envs/xgbost-env.yaml
[OPEN-CE-ERROR-10] Error creating Build Tree
[OPEN-CE-ERROR-0] Unexpected Error: Error in https://raw.githubusercontent.com/open-ce/open-ce-environments/main/envs/xgbost-env.yaml:
  [OPEN-CE-ERROR-25] Failed to download https://raw.githubusercontent.com/open-ce/open-ce-environments/main/envs/xgbost-env.yaml with error:
HTTP Error 404: Not Found
```

File: open_ce/container_build.py [hunk 1]
```diff
@@ -222,14 +223,7 @@ def build_with_container_tool(args, arg_strings):
...
 
-    # env_config_file being positional argument cause problem while parsing known
-    # arguments. Hence removing it from arg_strings, as it is anyway being read
-    # from args ahead.
-    for env_file in args.env_config_file:
-        if env_file in arg_strings:
-            arg_strings.remove(env_file)
-
-    parser = make_parser()
+    parser = make_parser(len(args.env_config_file))
```
Comment (jayfurmanek) [inline open_ce/container_build.py:226]: I'm glad this is a little cleaner now.

File: open_ce/inputs.py [hunk 2]
```diff
@@ -256,6 +261,36 @@ def add_subparser(subparsers, command, arguments, *args, formatter_class=OpenCEF
...
     return subparser
 
+def _create_env_config_paths(args):
+    '''
+    If the provided env_config_file's don't exist locally, convert the paths to
+    URLs pointing to the GitHub repository for environemnt files.
+    '''
+    if "env_config_file" in vars(args).keys():
+        for index, config_file in enumerate(args.env_config_file):
+            if not os.path.exists(config_file) and not utils.is_url(config_file):
```
Comment (jayfurmanek) [inline open_ce/inputs.py:271]: Should we do an: " INFO: Found config_file "  on the else here?

**Review by jayfurmanek — APPROVED** [review]
Looks good!

File: open_ce/container_build.py (hunk 1 above)
Comment (bnemanich) [reply open_ce/container_build.py:226]: Unfortunately, this didn't work. I'll be pushing up a new PR that adds a modified version of the original back in.

**Pull request merged**